### ISSUE TITLE
Clarify the example @AfterPermissionGranted(RC_CAMERA_AND_LOCATION) 

### ISSUE THREAD

**Issue by salamanders** [googlesamples/easypermissions#202]
In the README example

    @AfterPermissionGranted(RC_CAMERA_AND_LOCATION)
    private void methodRequiresTwoPermission() {
        String[] perms = {Manifest.permission.CAMERA, Manifest.permission.ACCESS_FINE_LOCATION};
        if (EasyPermissions.hasPermissions(this, perms)) {

Why is it checking for the permissions?  The method only gets called once the two permissions are granted, so isn't it safe at that point to assume everything is ok?

**Comment by tjohns** [comment]
Off the top of my head... it's good practice to check, because the user can always retroactively revoke settings from the Settings app.

**Comment by salamanders** [comment]
In that case I'm not understanding the `@AfterPermissionGranted(RC_CAMERA_AND_LOCATION)` annotation - Isn't that saying "call this method as soon as the permissions are granted, and not a moment sooner."?

**Comment by tjohns** [comment]
Good question. @samtstern might have some more insight into that.

**Comment by SUPERCILEX** [comment]
@salamanders yes, you are correct about how the annotation works. The idea is that you can call that method with or without permissions and it will take care of itself. If you already have the permissions, things will go through. If you don't, it'll wait for the `onPermissionGranted` callback and then go through.

TL;DR: you don't need to check for permissions if you never manually call that method. 👍

**Comment by samtstern** [comment]
Also thanks @tjohns for jumping in on easypermissions!
